### ISSUE TITLE
Incorrect compilation of dependent Angular libraries  error TS6059

### ISSUE THREAD

**Issue by simonstone97** [nrwl/nx#8716]
## Current Behavior
With the update to Nrwl version 13.4.6 I could not compile my project correctly. In this project there are numerous Angular libraries that are partially dependent on each other.
When trying to compile the first library that has a dependency on another. I get hundreds of error messages. A few examples of error messages from the test repo are listed below.
In a test repo I tried to show the scenario. I noticed that the compilation errors only happen in the folder of the original project. Even if the content or the whole project is identical to one in another folder.  
I could only reproduce the error in the test repo by deleting the hidden folder `.git` from the directory.

I hope the error I was able to reproduce is identical to the error in my original project. Although the original project still has the `.git` folder in the directory, the behavior during compilation is identical to the behavior in the test repo when the `git` folder is no longer present.

Edit: New insights: 
there is no `.git` folder in the original project, because the project is part of a large monorepo and therefore only a subfolder. The `.git` folder is located in the overlying directory.

## Steps to Reproduce
Test Repo to readjust the error: https://github.com/simonstone97/nx-broken-libs-test

1. repo cloning
2. run `npm install`
3. run `npm run build:libs` This command tries to compile both libraries
4. the compilation process should have been successful
5. show hidden files in the directory and delete the folder `.git`
6. the compilation process fails with the error messages listed below
7. the `.git` folder can be added again and all steps from step 2 can be repeated with nx version `13.4.5` where the error does not occur

### Failure Logs
```
libs/lib-one/src/lib/models/theme/model.ts:1:1 - error TS6059: File 'C:/devtest/Nrwl/nx-broken-libs-test/libs/lib-one/src/lib/models/theme/model.ngtypecheck.ts' is not under 'rootDir' 'C:\devtest\Nrwl\nx-broken-libs-test\libs\lib-two\src'. 'rootDir' is expected to contain all source files.

1 export interface TestModel {

libs/lib-one/src/lib/models/models.export.ts:1:1 - error TS6059: File 'C:/devtest/Nrwl/nx-broken-libs-test/libs/lib-one/src/lib/models/models.export.ngtypecheck.ts' is not under 'rootDir' 'C:\devtest\Nrwl\nx-broken-libs-test\libs\lib-two\src'. 'rootDir' is expected to contain all source files.

1 import {TestModel} from './theme/model';

libs/lib-one/src/lib/models/models.export.ts:1:25 - error TS6059: File 'C:/devtest/Nrwl/nx-broken-libs-test/libs/lib-one/src/lib/models/theme/model.ts' is not under 'rootDir' 'C:\devtest\Nrwl\nx-broken-libs-test\libs\lib-two\src'. 'rootDir' is expected to contain all source files.

1 import {TestModel} from './theme/model';
                          ~~~~~~~~~~~~~~~
libs/lib-one/src/index.ts:1:1 - error TS6059: File 'C:/devtest/Nrwl/nx-broken-libs-test/libs/lib-one/src/index.ngtypecheck.ts' is not under 'rootDir' 'C:\devtest\Nrwl\nx-broken-libs-test\libs\lib-two\src'. 'rootDir' is expected to contain all source files.

1 export * from './lib/models/models.export';

libs/lib-one/src/index.ts:1:15 - error TS6059: File 'C:/devtest/Nrwl/nx-broken-libs-test/libs/lib-one/src/lib/models/models.export.ts' is not under 'rootDir' 'C:\devtest\Nrwl\nx-broken-libs-test\libs\lib-two\src'. 'rootDir' is expected to contain all source files.

1 export * from './lib/models/models.export';
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libs/lib-two/src/lib/sub-directory/values.ts:2:27 - error TS6059: File 'C:/devtest/Nrwl/nx-broken-libs-test/libs/lib-one/src/index.ts' is not under 'rootDir' 'C:\devtest\Nrwl\nx-broken-libs-test\libs\lib-two\src'. 'rootDir' is expected to contain all source files.

2 import { TestModel } from 'lib-one';
                            ~~~~~~~~~
```

### Environment
```
  Node : 16.13.0
  OS   : win32 x64
  npm  : 8.3.0

  nx : 13.4.5
  @nrwl/angular : 13.4.5
  @nrwl/cli : 13.4.5
  @nrwl/cypress : 13.4.5
  @nrwl/devkit : 13.4.5
  @nrwl/eslint-plugin-nx : 13.4.5
  @nrwl/express : undefined
  @nrwl/jest : 13.4.5
  @nrwl/linter : 13.4.5
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : undefined
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.4.5
  @nrwl/web : undefined
  @nrwl/workspace : 13.4.5
  @nrwl/storybook : 13.4.5
  @nrwl/gatsby : undefined
  typescript : 4.5.5
  rxjs : 7.4.0
  ---------------------------------------
  Community plugins:
         @angular/animations: 13.1.3
         @angular/cdk: 13.1.3
         @angular/common: 13.1.3
         @angular/compiler: 13.1.3
         @angular/core: 13.1.3
         @angular/forms: 13.1.3
         @angular/platform-browser: 13.1.3
         @angular/platform-browser-dynamic: 13.1.3
         @angular/router: 13.1.3
         @angular/service-worker: 13.1.3
         @angular-devkit/build-angular: 13.1.4
         @angular/cli: 13.1.4
         @angular/compiler-cli: 13.1.3
         @angular/language-service: 13.1.3

(node:6440) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./" in the "exports" field module resolution of the package at C:\devtest\Nrwl\nx-broken-libs-test\node_modules\tslib\package.json.
Update this package.json to use a subpath pattern like "./*".
(Use `node --trace-deprecation ...` to show where the warning was created)
```

**Comment by FrozenPandaz** [comment]
Thank you for providing a repro! Unfortunately, I pulled it down and ran the following:

```
npm i # Installs Nx 13.5.3
npm run build:libs # Success
rm -rf .git
npm run build:libs # Still Success
```

**Comment by simonstone97** [comment]
The behavior is seen in a Windows 10 environment. You have tested it on Linux or ? 
Please try it on a Windows device.

**Comment by simonstone97** [comment]
Since the update to 13.7.1 the compilation process works again. 
I think this has something to do with the fix #8764.
